dcicutils/zip_utils.py

```diff
@@ -0,0 +1,79 @@
+from contextlib import contextmanager
+import gzip
+import os
+import shutil
+import tarfile
+import tempfile
+from typing import List, Optional, Union
+import zipfile
+
+
+@contextmanager
+def unpack_zip_file_to_temporary_directory(file: str) -> str:
+    with temporary_directory() as tmp_directory_name:
+        with zipfile.ZipFile(file, "r") as zipf:
+            zipf.extractall(tmp_directory_name)
+        yield tmp_directory_name
+
+
+@contextmanager
+def unpack_tar_file_to_temporary_directory(file: str) -> str:
+    with temporary_directory() as tmp_directory_name:
+        with tarfile.open(file, "r") as tarf:
+            tarf.extractall(tmp_directory_name)
+        yield tmp_directory_name
+
```

Comment:
You might want to allow optional args to these functions in case additional options are needed/desired